src/Tizen.Uix.Sticker/Tizen.Uix.Sticker/StickerProvider.cs

```diff
@@ -0,0 +1,301 @@
+                Initialized = true;
+            }
+            else
+            {
+                Log.Debug(LogTag, "Already initialized");
+            }
+        }
+
+        /// <summary>
+        /// Deinitialize the sticker provider.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        /// <feature>http://tizen.org/feature/ui_service.sticker</feature>
+        /// <exception cref="NotSupportedException">The required feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">This exception can be due to operation failed.</exception>
+        /// <pre>
+        /// StickerProvider must be initialized.
+        /// </pre>
+        public static void Deinitialize()
+        {
+            ErrorCode error = StickerProviderDestroy(_handle);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "Destroy Failed with error " + error);
+                if (error == ErrorCode.InvalidParameter)
+                    throw ExceptionFactory.CreateException(ErrorCode.OperationFailed);
+                else
+                    throw ExceptionFactory.CreateException(error);
+            }
+        }
+
+        /// <summary>
+        /// Inserts a sticker data to the sticker database.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        /// <feature>http://tizen.org/feature/ui_service.sticker</feature>
+        /// <param name="data">The sticker data to be saved.</param>
+        /// <exception cref="NotSupportedException">The required feature is not supported.</exception>
+        /// <exception cref="ArgumentException">This exception can be due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">
+        /// This can occur due to the following reasons:
+        /// 1) This exception can be due to operation failed.
+        /// 2) This exception can be due to file exists.
+        /// 3) This exception can be due to no such file.
+        /// </exception>
+        public static void InsertData(StickerData data)
+        {
+            ErrorCode error = StickerProviderInsertData(_handle, data.SafeStickerDataHandle);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "InsertData Failed with error " + error);
+                throw ExceptionFactory.CreateException(error);
+            }
+        }
+
+        /// <summary>
+        /// Inserts a sticker data using JSON file.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        /// <feature>http://tizen.org/feature/ui_service.sticker</feature>
+        /// <remarks>
+        /// All data except thumbnail, description, and display_type must be set in the JSON file to insert the sticker data.
+        /// @a jsonPath must have a non-null value and must be an existing file.
+        /// If the URI type is lacal path, the sticker file is copied to a sticker directory.
```

Comment:
what URI type? I see no URI type in the signature. Do you mean the proto part file:// 🆚 http:// ?

Reply:
URI type is used to distinguish whether the sticker file exists in the app package directory or a web resource.